### PR TITLE
Drop duplicate indices

### LINKED ISSUE #58648: ENH: Add paramenter `index` to `drop_duplicates` to drop duplicate indices
### Feature Type

- [ ] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

There currently is no elegant pattern to drop duplicate indices.  
I think what people usually do is 
`df[~df.index.duplicated(keep='first')]`

### Feature Description

Add a new parameter to `drop_duplicates` to specify dropping duplicate indices.  
An option could be a `index=True` to do this, similar to when merging on an index.

### Alternative Solutions

Allow the `subset` parameter of `drop_duplicates` to accept the name of the index.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Yousinator**
- [x] closes #58648
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Tests are added but have not been run, as I get this error when I try to run them. I tried all envorment setup options but I end-up having the same error

```shell
root@92c4f0dec9a0:/home/pandas# pytest pandas/tests/frame/methods/test_drop_duplicates.py
ImportError while loading conftest '/home/pandas/pandas/conftest.py'.
pandas/__init__.py:47: in <module>
    from pandas.core.api import (
pandas/core/api.py:46: in <module>
    from pandas.core.groupby import (
pandas/core/groupby/__init__.py:1: in <module>
    from pandas.core.groupby.generic import (
pandas/core/groupby/generic.py:1418: in <module>
    class DataFrameGroupBy(GroupBy[DataFrame]):
pandas/core/groupby/generic.py:2571: in DataFrameGroupBy
    @doc(DataFrame.plot.__doc__)
E   AttributeError: type object 'DataFrame' has no attribute 'plot'
```

### CODE REVIEW

**Comment by Yousinator** [comment]
Most of these tests are failing because of the same error mentioned earlier
```bash
AttributeError: type object 'DataFrame' has no attribute 'plot'
```

others like precommit fail from parts that I haven't edited

File: pandas/core/frame.py [hunk 1]
```diff
@@ -6546,109 +6544,133 @@ def drop_duplicates(
...
-        inplace : bool, default ``False``
-            Whether to modify the DataFrame rather than creating a new one.
-        ignore_index : bool, default ``False``
-            If ``True``, the resulting axis will be labeled 0, 1, …, n - 1.
 
-        Returns
-        -------
-        DataFrame or None
-            DataFrame with duplicates removed or None if ``inplace=True``.
+def drop_duplicates(
```
Comment (yuanx749) [inline pandas/core/frame.py:6548]: You have dedented the method `drop_duplicates`.

Comment (Yousinator) [reply pandas/core/frame.py:6548]: it seems my code decided to take a step back 😶‍🌫️
 I'll get it back in line. Thanks for catching that!

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**